ds3-sdk-samples/src/main/java/com/spectralogic/ds3client/samples/RecoverJobExample.java

```diff
@@ -0,0 +1,101 @@
+        try (final Ds3Client client = Ds3ClientBuilder.fromEnv().withHttps(false).build()) {
+            final String bucketName = "recover_get_books_job_bucket";  //The bucket we are interested in getting objects from.
+            Ds3ClientHelpers helper = Ds3ClientHelpers.wrap(client);
+            helper.ensureBucketExists(bucketName);
+
+            // Our local path which contains all the files that we want to transfer
+            // This example assumes that there is at least 2 files in the "input" directory
+            final Path inputPath = Paths.get("input");
+
+            // Get the list of files that are contained in the inputPath
+            final Iterable<Ds3Object> objects = helper.listObjectsForDirectory(inputPath);
+
+            // Create the write job with the bucket we want to write to and the list
+            // of objects that will be written
+            final Ds3ClientHelpers.Job job = helper.startWriteJob(bucketName, objects);
+
+            // Start the write job using an Object Putter that will read the files
+            // from the local file system.
+            job.transfer(new FileObjectPutter(inputPath));
+
+            // Create a local output directory to place retrieved objects into
+            final Path downloadPath = FileSystems.getDefault().getPath("output");
+            if (!Files.exists(downloadPath)) {
+                Files.createDirectory(downloadPath);
+            }
+
+            // Get the first object
+            final List<Ds3Object> objectsList = Lists.newArrayList(objects);
+            final Ds3Object object1 = objectsList.get(0);
+            final FileChannel channel1 = FileChannel.open(
+                    downloadPath.resolve(object1.getName()),
+                    StandardOpenOption.WRITE,
+                    StandardOpenOption.CREATE,
+                    StandardOpenOption.TRUNCATE_EXISTING
+            );
+            final Ds3ClientHelpers.Job readJob = helper.startReadJob(bucketName, objectsList);
+            client.getObject(new GetObjectRequest(bucketName, object1.getName(), 0, readJob.getJobId(), channel1));
+
+            /**
+             * Here is where we attempt to recover from a hypothetical interruption - before we get the 2nd object,
+             * and while the job is still "In Progress"
+             */
+            try {
+                final Ds3ClientHelpers.Job recoverJob = helper.recoverReadJob(readJob.getJobId());
+                final Ds3Object object2 = objectsList.get(1);
+                final FileChannel channel2 = FileChannel.open(
+                        downloadPath.resolve(object2.getName()),
+                        StandardOpenOption.WRITE,
+                        StandardOpenOption.CREATE,
+                        StandardOpenOption.TRUNCATE_EXISTING
+                );
+
+                client.getObject(new GetObjectRequest(bucketName, object2.getName(), 0, recoverJob.getJobId(), channel2));
```

Comment:
Instead of getObject, it may be good to show them that they can use the transfer method here.